infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/infrastructure/CorfuReplicationDiscoveryService.java

```diff
@@ -706,18 +708,33 @@ public Map<LogReplicationSession, ReplicationStatus> queryReplicationStatus() {
             return replicationStatusMap;
         }
 
-        if (topologyDescriptor.getLocalClusterDescriptor().getRole() != ClusterRole.SOURCE &&
-                topologyDescriptor.getLocalClusterDescriptor().getRole() != ClusterRole.SINK) {
-            log.error("Replication Status Query in incorrect role {}", topologyDescriptor.getLocalClusterDescriptor().getRole());
+        if (!isSource() && !isSink()) {
+            log.error("Received Replication Status Query in Incorrect Role, cluster is neither SOURCE/SINK");
             return replicationStatusMap;
         }
 
         return sessionManager.getReplicationStatus();
     }
 
+
+    /**
+     * Called by clients when needed to enforce a snapshot sync for a session.
+     *
+     * @param session against which the local cluster has to enforce a snapshot sync
+     * @return event ID
+     * @throws LogReplicationDiscoveryServiceException
+     */
     public UUID forceSnapshotSync(LogReplicationSession session) throws LogReplicationDiscoveryServiceException {
-        if(topologyDescriptor.getLocalClusterDescriptor().getRole() == ClusterRole.SINK) {
-            String errorStr = "Force snapshot sync not supported on sink cluster.";
+        if (!topologyDescriptor.getLocalClusterDescriptor().getClusterId().equals(session.getSourceClusterId())) {
```

Comment:
when can we hit this condition?

Reply:
Anny pointed out in https://github.com/CorfuDB/CorfuDB/pull/3510#discussion_r1095165678 that this API can be called externally.
We can have a gating on the client side, but its good to have it here as well..